scripts/externalTests/test_helpers.py

```diff
@@ -0,0 +1,284 @@
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2023 solidity contributors.
+# ------------------------------------------------------------------------------
+
+import os
+import re
+import subprocess
+import sys
+from abc import ABCMeta
+from argparse import ArgumentParser
+from dataclasses import dataclass, field
+from pathlib import Path
+from shutil import rmtree
+from tempfile import mkdtemp
+from textwrap import dedent
+from typing import List
+
+# Our scripts/ is not a proper Python package so we need to modify PYTHONPATH to import from it
+# pragma pylint: disable=import-error,wrong-import-position
+PROJECT_ROOT = Path(__file__).parents[2]
+sys.path.insert(0, f"{PROJECT_ROOT}/scripts/common")
+
+from git_helpers import git_commit_hash
+
+SOLC_FULL_VERSION_REGEX = re.compile(r"^[a-zA-Z: ]*(.*)$")
+SOLC_SHORT_VERSION_REGEX = re.compile(r"^([0-9.]+).*\+|\-$")
+
+evm_version = os.environ.get("DEFAULT_EVM")
+CURRENT_EVM_VERSION: str = evm_version if evm_version is not None else "shanghai"
+
+AVAILABLE_PRESETS: List[str] = [
+    "legacy-no-optimize",
+    "ir-no-optimize",
+    "legacy-optimize-evm-only",
+    "ir-optimize-evm-only",
+    "legacy-optimize-evm+yul",
+    "ir-optimize-evm+yul",
+]
+
+
+@dataclass
+class TestConfig:
+    name: str
+    repo_url: str
+    ref_type: str
+    ref: str
+    build_dependency: str = field(default="nodejs")
+    compile_only_presets: List[str] = field(default_factory=list)
+    settings_presets: List[str] = field(default_factory=lambda: AVAILABLE_PRESETS)
+    evm_version: str = field(default=CURRENT_EVM_VERSION)
+
+    def selected_presets(self):
+        return set(self.compile_only_presets + self.settings_presets)
+
+
+class TestRunner(metaclass=ABCMeta):
+    config: TestConfig
+    solc_binary_type: str
+    solc_binary_path: Path
+
+    def __init__(self, argv, config: TestConfig):
+        args = parse_command_line(f"{config.name} external tests", argv)
+        self.config = config
+        self.solc_binary_type = args.solc_binary_type
+        self.solc_binary_path = args.solc_binary_path
+        self.env = os.environ.copy()
+        self.tmp_dir = mkdtemp(prefix=f"ext-test-{config.name}-")
+        self.test_dir = Path(self.tmp_dir) / "ext"
+
+    def setup_solc(self) -> str:
+        if self.solc_binary_type == "solcjs":
+            # TODO: add support to solc-js # pylint: disable=fixme
+            raise NotImplementedError()
+        print("Setting up solc...")
+        solc_version_output = subprocess.getoutput(f"{self.solc_binary_path} --version").split(":")[1]
```

Comment:
Doesn't this ignore the exit code?

Also, we don't really need a shell here.

```suggestion
        solc_version_output = subprocess.check_output([self.solc_binary_path, '--version']).split(":")[1]
```

Reply:
Yes it does. I will use `check_output` instead.